src/transaction_signature_options/mod.rs

```diff
@@ -119,96 +119,160 @@ impl interactive_clap::FromCli for Submit {
-                            optional_clap_variant,
-                            color_eyre::Report::msg(report),
-                        )
-                    }
-                };
-                match (context.on_after_sending_transaction_callback)(
-                    &transaction_info,
-                    &context.network_config,
-                ) {
-                    Ok(_) => (),
-                    Err(report) => {
-                        return interactive_clap::ResultFromCli::Err(
-                            optional_clap_variant,
-                            color_eyre::Report::msg(report),
-                        )
-                    }
-                };
-                eprintln!("{storage_message}");
-                interactive_clap::ResultFromCli::Ok(CliSubmit::Send)
-            }
-            Some(CliSubmit::Display) => {
-                match (context.on_before_sending_transaction_callback)(
-                    &context.signed_transaction,
-                    &context.network_config,
-                    &mut storage_message,
-                ) {
-                    Ok(_) => (),
-                    Err(report) => {
-                        return interactive_clap::ResultFromCli::Err(
-                            optional_clap_variant,
-                            color_eyre::Report::msg(report),
+                    match crate::common::print_transaction_status(
+                        &transaction_info,
+                        &context.network_config,
+                    ) {
+                        Ok(_) => (),
+                        Err(report) => {
+                            return interactive_clap::ResultFromCli::Err(
+                                optional_clap_variant,
+                                color_eyre::Report::msg(report),
+                            )
+                        }
+                    };
+                    match (context.on_after_sending_transaction_callback)(
+                        &transaction_info,
+                        &context.network_config,
+                    ) {
+                        Ok(_) => (),
+                        Err(report) => {
+                            return interactive_clap::ResultFromCli::Err(
+                                optional_clap_variant,
+                                color_eyre::Report::msg(report),
+                            )
+                        }
+                    };
+                    eprintln!("{storage_message}");
+                    interactive_clap::ResultFromCli::Ok(CliSubmit::Send)
+                }
+                SignedTransactionOrSignedDelegateAction::SignedDelegateAction(
+                    signed_delegate_action,
+                ) => {
+                    let client = reqwest::blocking::Client::new();
+                    let payload = signed_delegate_action.try_to_vec().unwrap();
+                    let json_payload = serde_json::to_vec(&payload).unwrap();
```

Comment:
Note to myself, we should better use the following payload:

```json
{
  "signed_delegate_action": "base64-encoded-string-of-borsh-serialized-signed-delegate-action"
}
```

, instead of the current array of numbers in JSON of the borsh-serialized signed delegate action:

```json
[123, 125, 254, 23, 23, 34, ...]
```